lib/grandpa/network.go

```diff
@@ -33,6 +31,7 @@ var (
 	grandpaID                protocol.ID = "/paritytech/grandpa/1"
 	messageID                            = network.ConsensusMsgType
 	neighbourMessageInterval             = time.Minute * 5
+	gossipMessage                        = scale.MustNewVaryingDataType(&VoteMessage{}, &CommitMessage{}, &NeighbourMessage{}, &catchUpRequest{}, &catchUpResponse{})
```

Comment:
don't use the same VaryingDataType in `decodeMessage`.  This isn't thread safe if there are parallel calls to `decodeMessage`. You will be overwriting the same address repeatedly, which could lead to undesired side effects.